horizon/enforcer/utils/mapping_rules_utils.py

```diff
@@ -96,14 +142,36 @@ def extract_mapping_rule_by_request(
         url: AnyHttpUrl,
     ) -> MappingRuleData | None:
         matched_mapping_rules = []
+        http_method = http_method.lower()  # Convert once instead of in each iteration
+
         for mapping_rule in mapping_rules:
-            if mapping_rule.http_method != http_method.lower():
+            is_regex = mapping_rule.url_type == UrlTypes.REGEX
+
+            logger.debug(
+                "Checking mapping rule",
+                rule_method=mapping_rule.http_method,
+                request_method=http_method,
+                rule_url=mapping_rule.url,
+                request_url=str(url),
+                is_regex=is_regex,
+            )
```

Comment:
should be 
logger.debug(
                "checking mapping rule",
                rule_url=mapping_rule.url,
                rule_method=mapping_rule.http_method,
                rule_type=getattr(mapping_rule, "url_type", None),
                request_url=url,
                request_method=http_method,
                is_regex=is_regex,
            )